network/endpoint_windows.go

```diff
@@ -281,14 +282,28 @@ func (nw *network) configureHcnEndpoint(epInfo *EndpointInfo) (*hcn.HostComputeE
 }
 
 func (nw *network) deleteHostNCApipaEndpoint(cli apipaClient, networkContainerID string) error {
-	log.Printf("[net] Deleting HostNCApipaEndpoint for network container: %s", networkContainerID)
-	err := cli.DeleteHostNCApipaEndpoint(context.TODO(), networkContainerID)
-	log.Printf("[net] Completed HostNCApipaEndpoint deletion for network container: %s"+
-		" with error: %v", networkContainerID, err)
+	// First try to delete the host NC apipa endpoint directly from CNI
+	log.Printf("[net] Deleting HosNCApipaEndpoint for network container [%s] directly")
+	if err := nw.deleteHostNCApipaEndpointDirect(networkContainerID); err != nil {
+		log.Printf("[net] Error deleting HostNCApipaEndpoint for network container [%s] directly, fallback to CNS")
+		// Fallback to deleting the host NC apipa endpoint with CNS
+		log.Printf("[net] Deleting HostNCApipaEndpoint for network container [%s] with CNS", networkContainerID)
```

Comment:
why fallback to cns? if it fails, then cns also does the same right?

Reply:
removed